src/test/java/Tests/Task1_Tests.java

```diff
@@ -1,133 +1,31 @@
-        Assert.assertTrue(deleteButton.isDisplayed());
-        Thread.sleep(1500);
-
-        deleteButton.click();
-        Thread.sleep(1500);
-    }
-
-    public void userRegistrationWithValidData() throws InterruptedException {
-        Actions actions = new Actions(driver);
-
-//        WebElement summaryQuantity = driver.findElement((By.xpath("//span[@class='summary-quantity']")));
-        String amountOfProduct = "50";
-        WebElement google = driver.findElement(By.xpath("//h4[text()='Okulary']"));
-
-        WebElement inputGooglePrice = driver.findElement(By.xpath("//h4[text()='Piłka']/following-sibling::div[@class='input-group input-group-sm']//input"));
-        WebElement inputHeadSet = driver.findElement(By.xpath("//h4[text()='Słuchawki']/following-sibling::div[@class='input-group input-group-sm']//input"));
-
-        actions.moveToElement(inputHeadSet).build().perform();
-//        actions.moveToElement(google).clickAndHold().perform();
-
-        Thread.sleep(5000);
-
-
-        inputGooglePrice.click();
-        inputGooglePrice.clear();
-        inputGooglePrice.sendKeys(amountOfProduct);
-
-//        driver.switchTo().defaultContent();
-
-        Thread.sleep(5000);
-
-        inputHeadSet.click();
-        inputHeadSet.clear();
-        inputHeadSet.sendKeys(amountOfProduct);
-
-        Thread.sleep(5000);
-//        WebElement addButtonGoogle = driver.findElement(By.xpath("//button[@id='add-product-5d092e4a64bc5']"));
-        WebElement pilka = driver.findElement(By.id("add-product-5d07df3a2d1fe"));
-
-        driver.switchTo().frame(pilka);
-
-        pilka.isDisplayed();
-        pilka.isEnabled();
-        pilka.click();
-
-        Thread.sleep(5000);
-//
-//        actions.moveToElement(addButtonGoogle).build().perform();
-//        addButtonGoogle.click();
-//
-//        Thread.sleep(5000);
-
-
-
-
-//        WebElement inputGooglePrice = driver.findElement(By.xpath("//h4[text()='Okulary']/following-sibling::div[@class='input-group input-group-sm']//input"));
-
-//        Thread.sleep(5000);
-//        inputGooglePrice.click();
-//        inputGooglePrice.clear();
-//        inputGooglePrice.sendKeys(amountOfProduct);
-
-//        this.searchElementInputAndTypeValue(inputGooglePrice, amountOfProduct);
-//        actions.moveToElement(addButtonGoogle).perform();
-//        addButtonGoogle.click();
-//        this.tapAddButton(addButtonGoogle);
-//        String basketSummary = summaryQuantity.getText();
-//        Assert.assertEquals(basketSummary, summaryQuantity);
-
-
-//        WebElement cenaGoogle = google.findElement(By.xpath("/following-sibling::p[text()='Cena: 15.54 zł']"));
-
-
-
+        homePage.openTaskFirst();
+        taskOnePage.checkTaskOneHeaderisDisplayed();
+        WebElement addButton = taskOnePage.clickAddButton();
+        WebElement deleteButton = taskOnePage.clickDeleteButton();
+        Assert.assertTrue(addButton.isDisplayed());
+        Assert.assertFalse(!deleteButton.isDisplayed());
```

Comment:
1. po co ten wykrzyknik?
2. dwa asserty w jednym teście?
3. nazwa funkcji sugeruje, że testujesz prawidłowość dodawania i uswania elementów, tymczasem asserty jakby troszkę co innego robiły...?

Reply:
1. wykrzyknik to pomyłka - usunięty;
2. testuję prawidłowość dodania i usuwania elementów, w momencie kiedy dodamy element powinien pokazać się deleteButton (linia 28 to sprawdza, gdyby go nie było test by nie przeszedł bo nie bylibyśmy w stanie kliknąć), następnie go klikam (ze względu, że dodałam jeden element i następnie go usunęłam, nie ma żadnych innych elementów do usunięcia więc deleteButton nie istnieje - stąd linia 30). Add Button powinien być widoczny cały czas więc dla pewności na koniec testu sprawdzam czy po usunięciu wciąż jest dostępny.

Reply:
Nadal - test sprawdza czy wyświetlone są przyciski, a nie czy prawidłowo zadziałało dodawanie elementów ;). przemyśl.